### ISSUE TITLE
[Chore] Improve the logic to parse, store & enqueu the keywords

### ISSUE THREAD

**Issue by longnd** [aungkoko1234/web-scrapper-backend#21]
Issue
---
Currently, after the keywords list is parsed, the entire list is enqued
https://github.com/aungkoko1234/web-scrapper-backend/blob/9c520dc93598646fcc0f72d54845ec7a2838a985/src/keywords/keywords.controller.ts#L85-L97

then, the KeywordProcessor starts to
- craw the result for each keyword
- insert the keyword & their crawing result to the DB
https://github.com/aungkoko1234/web-scrapper-backend/blob/9c520dc93598646fcc0f72d54845ec7a2838a985/src/keywords/keywords.processor.ts#L25-L34

by doing that, there're the following consequences
- each keyword is blocked from inserting to DB until the previous keyword is completed the crawling process. The user will not see the complete list of keywords until all of them are fully crawled. 
- If something went wrong during the crawling process of a keyword, the job crash, the remaining keywords will not be inserted to DB.

isn't it better to 
- parse the list of keyword
- insert all of them to DB
- enque each keyword to the queue, also having a status column with value set as `Initialized` 
- the background job
  -  pick up each keyword from the queue
  -  update the crawling status to `processing`
  - and update the crawling result to DB
  - update the status to `completed`

by doing so, the user can see the list of keyword right after they upload, they can also see the status of each keyword

**Comment by aungkoko1234** [comment]
Yeah...you are right. It should be handled like the process you mentioned.

**Comment by aungkoko1234** [comment]
@longnd  When I checked it again, inserting the keywords into db before queuing  is right. But I think we don't need to enqueue each keyword list with separate queue.  Because we do not crawl keyword from multiple pages in google and just crawl search data from google search page. We just want to crawl the google search with around 100 keywords. Searching single keyword from google search page takes not much time and processing. So I think I do not need to separate to enqueue each keyword with a queue.

**Comment by aungkoko1234** [comment]
I have improved the process of scrapping in this pull request #26 @longnd

**Comment by longnd** [comment]
IMO, it does not change much, with the latest changes, what the application does:
1. parse the list of keyword
2. enqueue each keyword separately, i.e. one by one
3. the background job will
  - pick up a keyword
  - insert it into DB
  - search and extract the result
  - update the status & search result for that keyword in DB

as the keywords are only inserted to DB by the background job, the entire keyword list will not be available to show to the user until all of the keywords are processed.
Note that the next keyword will be blocked from inserting to DB until the previous keyword in the queue is processed. 
That also makes the `status` column less useful.
If each keyword is inserted into DB before enqueuing, the list will be available to show to the user with the status changed by the background job.

**Comment by aungkoko1234** [comment]
I have moved the process to inserting all the keyword first in this pr #26
